frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/bc.ts

```diff
@@ -6,52 +9,80 @@ export enum Tier {
   SPREAD = 'SPREAD',
   MIRROR = 'MIRROR',
 }
-
 const TierCountMap = Object.freeze({
   [Tier.SPREAD]: 1,
   [Tier.MIRROR]: 2,
 });
 
-export const cleanup = () => {
-  cy.exec(
-    'oc delete backingstore test-store1 test-store2 test-store3 test-store4 -n openshift-storage',
-    {
-      timeout: 60000,
-    },
-  );
-};
+export enum BucketClassType {
+  STANDARD = 'STANDARD',
+  NAMESPACE = 'NAMESPACE',
+}
+
+export enum NamespacePolicyType {
+  SINGLE = 'Single',
+  MULTI = 'Multi',
+  CACHE = 'Cache',
+}
+
+abstract class BucketClassConfig {
+  public abstract setup: () => void;
+
+  public abstract cleanup: () => void;
+
+  constructor(public resources: string[], public type: BucketClassType) {}
+}
+
+export class StandardBucketClassConfig extends BucketClassConfig {
+  tiers: Tier[];
 
-const createPVCBackingStore = (storeName: string) => {
-  cy.log(`Creating a Backing Store resource named ${storeName}`);
-  const bucketStore = {
-    apiVersion: 'noobaa.io/v1alpha1',
-    kind: 'BackingStore',
-    metadata: {
-      name: storeName,
-    },
-    spec: {
-      pvPool: {
-        numVolumes: 1,
-        storageClass: 'gp2',
-        resources: {
-          requests: {
-            storage: '50Gi',
-          },
-        },
-      },
-      type: 'pv-pool',
-    },
+  private createPVCBackingStore = (storeName: string) => {
+    cy.log(`Creating a Backing Store resource named ${storeName}`);
+    cy.exec(
+      `echo '${JSON.stringify(bucketStore(storeName))}' | kubectl create -n openshift-storage -f -`,
+    );
   };
 
-  cy.exec(`echo '${JSON.stringify(bucketStore)}' | kubectl create -n openshift-storage -f -`);
-};
+  setup = () => this.resources.forEach(this.createPVCBackingStore);
 
-export const createBackingStore = () => {
-  createPVCBackingStore('test-store1');
-  createPVCBackingStore('test-store2');
-  createPVCBackingStore('test-store3');
-  createPVCBackingStore('test-store4');
-};
+  cleanup = () => {
+    cy.log('Deleting backing stores');
+    cy.exec(`oc delete backingstore ${this.resources.join(' ')} -n openshift-storage`);
+  };
+}
+
+export class NamespaceBucketClassConfig extends BucketClassConfig {
+  namespacePolicyType: NamespacePolicyType;
+
+  testBackingStore: string = 'backingstore-test';
```

Comment:
I dont think we want to modify this ?
```suggestion
  readonly testBackingStore: string = 'backingstore-test';
```

Reply:
for now !